lib/gala/payment_token.rb

```diff
@@ -20,14 +21,15 @@ def initialize(token_attrs)
       self.transaction_id = headers["transactionId"]
       self.ephemeral_public_key = headers["ephemeralPublicKey"]
       self.public_key_hash = headers["publicKeyHash"]
+      self.application_data = headers["applicationData"]
     end
 
     def decrypt(certificate_pem, private_key_pem)
+      self.class.validate_signature(signature, ephemeral_public_key, data, transaction_id, application_data)
```

Comment:
I see where you are coming form on this, but the apple documentation definitely views validation as the first step in the 5 step decryption process.  https://developer.apple.com/library/ios/documentation/PassKit/Reference/PaymentTokenJSON/PaymentTokenJSON.html
I am fine with breaking it out, just wanted to point out how Apple treats is as part of decryption.